### PR TITLE
Add mobility doc homepage and some other edits

### PR DESCRIPTION
**Pull request by rmglennon**
With the creation of https://mapzen.com/documentation/mobility/, the homepage should talk about Mobility, instead of just Turn-by-Turn. This creates a short page that mentions Turn-by-Turn, Optimized, and Matrix, + the existence of Transitland.

Other changes:
- add in multimodal where missing before
- remove some content on the repos in Valhalla
- take the content that was in the homepage and put it into a new file that is specifically about Turn-by-Turn

### CODE REVIEW

File: terminology.md [hunk 1]
```diff
@@ -1,20 +1,4 @@
...
-- [Thor](https://github.com/valhalla/thor) - Library used to generate a path through the graph tile hierarchy. This path can be used as input to `odin`. Depends on `midgard`, `baldr`, `sif`, `loki` and `odin`
-- [Odin](https://github.com/valhalla/odin) - Library used to generate maneuvers and narrative based on a path. This set of directions information can be used as input to `tyr`. Depends on `midgard` and `baldr`
-- [Tyr](https://github.com/valhalla/tyr) - Service used to handle http requests for a route communicating with all of the other valhalla APIs. The service will format output from `odin` and support json (and eventually protocol buffer) output. Depends on `midgard`, `baldr`, `sif`, `mjolnir`, `loki`, `thor` and `odin`
-- [Skadi](https://github.com/valhalla/skadi) - Library used to access digital elevation model data, which is useful in computing steepness of edges in the route graph or generating an elevation profile along a computed route.  Depends on `midgard` and `baldr`.
-
```
Comment (dnesbitt61) [inline terminology.md]: Are these present in another part of the Valhalla docs?

**Comment by rmglennon** [comment]
@dnesbitt61 - most of it is in the readme here https://github.com/valhalla/valhalla

We should combine anything removed from this file with the readme. They both look in need of updates, as libraries like meili are missing from both.

(All goes back to the challenge of making docs on the service vs the open-source project...hey, Q3 isn't quite over yet to make progress on that goal!)

**Comment by rmglennon** [comment]
Pulled in the formatting from #101

File: terminology.md [hunk 2]
```diff
@@ -1,37 +1,21 @@
...
+* __narration__ - textual guidance describing the maneuver to be performed, such as a turn, distance to travel, and expected time.
+* __path__ - the sequence of edges forming a route
+* __penalty__ - fixed costs in seconds that are only added to the path cost. Penalties can influence the route path determination but do not add to the estimated time along the path.
+* __route__ - sequence of edges and maneuvers forming the best travel path between locations given the available road network, costs, influence factors, and other inputs.
+* __short path__ - a route that attempts to minimize distance traveled over the constituent edges, but may not be the shortest distance.
+* __through location__ - an optional location to influence the route to travel through that location.
+* __tiled routing__ - method of building a path on graph data that has been split into square cells.
+* __time__ - the number of seconds estimated to complete a maneuver or trip, including any additional costs.
+* __trip__ - results of an entire route, including locations, legs, and maneuvers.
```
Comment (dgearhart) [inline terminology.md]: Maybe we should add `leg` to the definitions

**Comment by dnesbitt61** [comment]
:shipit:

**Pull request merged**